src/mlpack/tests/cv_test.cpp

```diff
@@ -164,6 +165,34 @@ BOOST_AUTO_TEST_CASE(MSETest)
   BOOST_REQUIRE_CLOSE(MSE::Evaluate(lr, data, responses), expectedMSE, 1e-5);
 }
 
+/**
+ * Test the R squared metric (R2Score).
+ */
+BOOST_AUTO_TEST_CASE(R2ScoreTest)
+{
+  // Making two points that define the linear function f(x) = x - 1
+  arma::mat trainingData("0 1");
+  arma::rowvec trainingResponses("-1 0");
+
+  LinearRegression lr(trainingData, trainingResponses);
+
+  // Making five responses that are the output of regression function f(x)
+  // with some responses having a slight deviation of 0.005
+  // Mean Responses = (1 + 2 + 3 + 6 + 8)/5 = 4
+  arma::mat data("2 3 4 7 9");
+  arma::rowvec responses("1 2.005 3 6.005 8.005");
+
+  double ss_reg = (0 * 0 + 0.005 * 0.005 + 0 * 0 +
```

Comment:
I think we can calculate the expectedR2 by hand instead of writing code to do it, since it is hardcoded

Reply:
That sounds good. Making changes. Thanks.